src/S3/MultipartCopy.php

```diff
@@ -107,19 +112,27 @@ protected function getUploadCommands(callable $resultHandler)
 
     private function createPart($partNumber, $partsCount)
     {
+        $data = [];
+
+        // Apply custom params to UploadPartCopy data
+        $config = $this->getConfig();
+        $params = isset($config['params']) ? $config['params'] : [];
+        foreach ($params as $k => $v) {
+            $data[$k] = $v;
```

Comment:
One potential problem with this approach is that it assumes parameters always have the same form and semantic meaning for each constituent command of a multipart operation. I believe that's true now (though I can't say for sure without looking at every parameter used by {Create,Complete,Abort}MultipartUpload/UploadPart), but is that something the can be relied on indefinitely? There's nothing in S3's API structure that guarantees that, which is I believe why multipart uploads must today be customized [by supplying `'before_X'` callbacks](http://docs.aws.amazon.com/aws-sdk-php/v3/guide/service/s3-multipart-upload.html#customizing-a-multipart-upload).

Take the following scenario: Say S3 adds a `foo` parameter to `CreateMultipartUpload` that takes a structure. Assume that it's a new kind of tagging. A couple months later, they release a feature that lets you apply additional `foo` to the individual parts of a multipart upload. Because `UploadPart` is a streaming operation, `foo` must be sent as a header, which means the SDK must receive it as a serialized string (like how the SDK takes `Tagging` as a string in [`PutObject`](http://docs.aws.amazon.com/aws-sdk-php/v3/api/api-s3-2006-03-01.html#putobject) (where it's a header) and as a structure in [`PutObjectTagging`](http://docs.aws.amazon.com/aws-sdk-php/v3/api/api-s3-2006-03-01.html#putobjecttagging) (where it's serialized as part of the request body)). After merging in this PR, that would be a breaking change for the PHP SDK, whereas today it wouldn't be.